src/wiringX.c

```diff
@@ -97,6 +80,144 @@ static struct spi_t spi[2] = {
 	} while (0)
 #endif
 
+#ifndef __FreeBSD__
+/* SPI Bus Parameters */
+struct spi_t {
+	uint8_t mode;
+	uint8_t bits_per_word;
+	uint16_t delay;
+	uint32_t speed;
+	int fd;
+} spi_t;
+
+/* SPI states list implementation */
+
+struct id_spi_t {
+	uint8_t device;
+	uint8_t channel;
+};
+
+struct l_spi_t {
+	struct id_spi_t id;
+	struct spi_t data;
+};
+
+struct l_spi_t *spistates = 0;
+int nrspistates = 0;
```

Comment:
Let's say your struct is 16bits. To make an array you allocate 16*5 bits for 5 array elements. That means that the size of the array is now 80bits. However, the size of each element of that array will still be 16 bits. So, 80 / 16 is again 5.

So it should be (i think):
`sizeof(spistates)/sizeof(struct spi_list_t)`.

Just try if you get the same result as the nrspistates int.